test/ecto/changeset_test.exs

```diff
@@ -1044,6 +1045,58 @@ defmodule Ecto.ChangesetTest do
     assert changeset.errors == [terms_of_service: {"must be abided", [validation: :acceptance]}]
   end
 
+  test "validate_unique_tentatively/3" do
+    defmodule FakeRepoWithDup do
```

Comment:
@josevalim I couldn't find (or maybe didn't recognize) what you were referring to here:

> One way to solve it is to use the process inbox or the process dictionary to instruct the adapter to return a certain value. We do so currently only on the migration tests.

... so this is my somewhat dumb solution. 😊  I'm happy to do it differently if someone has a better way.